tests/testthat/test-extractVariableGroups.R

```diff
@@ -0,0 +1,47 @@
+context("extractVariableGroups Test")
+
+test_that("check that extractVariableGroups correctly extracts variable groups", {
+  x1 <- c("a|+|1|+|aa",
+         "a|+|2|abc",
+         "a|+|1|+|bb",
+         "a|+|1|+|cc",
+         "a|+|3|+|aa",
+         "a|+|3|+|bb",
+         "a",
+         "a|+|3",
+         "a|++|bla",
+         "a|++|blu",
+         "a|+++|ble",
+         "a|+++|blet")
+
+  x2 <- c("a|+|1|+|aa",
+         "a|+|2|abc",
+         "a|+|1|+|bb",
+         "a|+|1|+|cc",
+         "a|+|3|+|aa",
+         "a|+|3|+|bb",
+         "a",
+         "a2",
+         "a|+|3",
+         "a2|+|bla",
+         "a2|+|blu",
+         "a3|++|ble",
+         "a3|++|blet")
+  
+  res1 <- list("a"=c("a|+|1|+|aa","a|+|2|abc","a|+|1|+|bb","a|+|1|+|cc","a|+|3|+|aa","a|+|3|+|bb","a|+|3"),
+              "a|+|1"=c("a|+|1|+|aa","a|+|1|+|bb","a|+|1|+|cc"),
+              "a|+|3"=c("a|+|3|+|aa","a|+|3|+|bb"),
+              "a 2"=c("a|++|bla","a|++|blu"),
+              "a 3"=c("a|+++|ble" ,"a|+++|blet"))
+
+  res2 <- list("a"=c("a|+|1|+|aa","a|+|2|abc","a|+|1|+|bb","a|+|1|+|cc","a|+|3|+|aa","a|+|3|+|bb","a|+|3"),
+               "a|+|1"=c("a|+|1|+|aa","a|+|1|+|bb","a|+|1|+|cc"),
+               "a|+|3"=c("a|+|3|+|aa","a|+|3|+|bb"),
+               "a2"=c("a2|+|bla","a2|+|blu"),
+               "a3"=c("a3|++|ble" ,"a3|++|blet"))
+  
+  expect_mapequal(extractVariableGroups(x1),res1)
+  expect_mapequal(extractVariableGroups(x2,keepOrigNames = T),res2)
```

Comment:
why do you perform the second test with x2 instead of x1? should work with x1 as well, right?

Reply:
no, `res1` has `a 2` and `a 3` as group names, these are the automatically generated names for groups whose total is missing

Reply:
here is the result that should come out for `keepOrigNames = TRUE` for ? `x1`:

```
> extractVariableGroups(x1, keepOrigNames = TRUE)
$a
[1] "a|+|1|+|aa" "a|+|2|abc"  "a|+|1|+|bb" "a|+|1|+|cc" "a|+|3|+|aa" "a|+|3|+|bb" "a|+|3"     

$`a|+|1`
[1] "a|+|1|+|aa" "a|+|1|+|bb" "a|+|1|+|cc"

$`a|+|3`
[1] "a|+|3|+|aa" "a|+|3|+|bb"

$a
[1] "a|++|bla" "a|++|blu"

$a
[1] "a|+++|ble"  "a|+++|blet"
```
Is that still the case?